generator/internal/predicate_utils.cc

```diff
@@ -134,6 +134,12 @@ bool HasRoutingHeader(MethodDescriptor const& method) {
   return absl::holds_alternative<HttpExtensionInfo>(result);
 }
 
+bool HasHttpAnnotation(google::protobuf::MethodDescriptor const& method) {
```

Comment:
I am curious about whether this needs to be separate from `HasRoutingHeader()`. It would help if I knew what the REST metadata decorator will look like. Is there a REST version of this code? https://github.com/googleapis/google-cloud-cpp/blob/bf41762a2af08f158f480f66d75df4223bccda84/generator/internal/metadata_decorator_generator.cc#L40-L44

Reply:
I am probably being dumb, and there just aren't routing instructions for the simple case.